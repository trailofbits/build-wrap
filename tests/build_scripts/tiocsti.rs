// smoelius: Based on: https://github.com/containers/bubblewrap/issues/142
// Thanks to @maxammann for the pointer.

macro_rules! i8_ptr {
    ($expr:expr) => {
        $expr as *const _ as *const i8
    };
}

fn main() {
    let cmd = "id\n";
    for c in cmd.chars() {
        if unsafe { libc::ioctl(libc::STDIN_FILENO, libc::TIOCSTI, i8_ptr!(&c)) } < 0 {
            unsafe { libc::perror(i8_ptr!(b"libc::ioctl\0")) };
            panic!();
        }
    }
}
