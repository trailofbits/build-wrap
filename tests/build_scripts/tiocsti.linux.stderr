...
  libc::ioctl: Inappropriate ioctl for device
...
